vmdb/app/models/service_template.rb

```diff
@@ -180,4 +180,21 @@ def self.default_provisioning_entry_point
   def self.default_retirement_entry_point
     '/Service/Retirement/StateMachines/ServiceRetirement/default'
   end
+
+  def template_valid?
+    return true if service_resources.size.zero?
+
+    if atomic?
+      msg_prefix = "Service Template: [#{name}]:"
+      sr  = service_resources.first
+      obj = sr.resource.source
+
+      return "#{msg_prefix} Unable to find VM with Id [#{sr.resource.source_id}]" if obj.nil?
+      return "#{msg_prefix} VM/Template <#{obj.name}> with Id <#{obj.id}> is archived and cannot be used with provisioning." if obj.archived?
+      return "#{msg_prefix} VM/Template <#{obj.name}> with Id <#{obj.id}> is orphaned and cannot be used with provisioning." if obj.orphaned?
```

Comment:
@lfu The logic for checking if the source is valid needs to be pushed down to the resource model, in this case miq_provision_request.  Then here we can loop over all the service_resources and check any that expose the `template_valid?` method.  With that you would not need the sub_services check below because the sub_services items are connected through the service_resources relationship.